x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.ts

```diff
@@ -519,6 +520,7 @@ export const typeSpecificCamelToSnake = (params: TypeSpecificRuleParams): Respon
         query: params.query,
         filters: params.filters,
         saved_id: params.savedId,
+        data_view_id: params.dataViewId,
```

Comment:
The fact that this didn't cause a type error before when it was missing exposes a flaw in the new rule schemas - the response schema allows fields to be truly optional, which makes it easy to forget to add optional fields here. I opened https://github.com/elastic/kibana/issues/136910 as a tech debt issue to improve the response schema and catch these bugs using the static type system.